influx/query.go

```diff
@@ -0,0 +1,45 @@
+package influx
+
+import (
+	"context"
+	"crypto/x509"
+	"fmt"
+
+	"github.com/apache/arrow/go/v12/arrow/flight"
+	"github.com/apache/arrow/go/v12/arrow/flight/flightsql"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/metadata"
+)
+
+func (c *Client) Query(ctx context.Context, bucket string, query string, queryParams interface{}) (*flight.Reader, error) {
+	pool, err := x509.SystemCertPool()
+	if err != nil {
+		return nil, fmt.Errorf("x509: %s", err)
+	}
+	transport := grpc.WithTransportCredentials(credentials.NewClientTLSFromCert(pool, ""))
+	opts := []grpc.DialOption{
+		transport,
+	}
+
+	url := ReplaceURLProtocolWithPort(c.configs.HostURL)
+
+	client, err := flightsql.NewClient(url, nil, nil, opts...)
+	if err != nil {
+		return nil, fmt.Errorf("flightsql: %s", err)
+	}
+	defer client.Close()

```

Comment:
Unhandled error from `Close()`:

```suggestion
	defer func(client *flightsql.Client) {
		err := client.Close()
		if err != nil {
			// deal with error
		}
	}(client)

```